src/hooks/useAllServices.ts

```diff
@@ -95,103 +100,114 @@ const filterAllServicesSections = (allServicesLinks: AllServicesSection[], filte
           // no links means all bundle requests have failed
-          error: Object.keys(availableLinks).length === 0,
+          error: availableLinks.flatMap(({ items }) => Object.keys(items)).length === 0,
         }));
       }
-    });
+    };
+    setNavigation();
     return () => {
       isMounted.current = false;
     };
-  }, []);
+  }, [fetchSections, fetchNavitation]);
 
-  // AllServices pages section
-  // update only on ready status change
-  // FIXME: Remove prod filtering once the data structure is outside of chrome
-  const linkSections = useMemo(() => {
-    // create a flat array of all available link href
-    const linksToMatch = allServicesLinks
-      .flatMap((item) => {
-        return item.links;
-      })
-      .flatMap((item) => {
-        // use router "path/*" to increase number of route matches
-        if (isAllServicesGroup(item)) {
-          // we have to filter items before the structure is offloaded outside of Chrome
-          return item.links.filter((item) => (isProdEnv ? item.prod !== false : true)).map(({ href }) => `${href}/*`);
-        }
-        if (isProdEnv && item.prod === false) {
-          // we have to filter items before the structure is offloaded outside of Chrome
-          return '';
-        }
-        return [`${item.href}/*`];
-      })
-      .filter((item) => item.length > 0);
-
-    // use router match to remove links that are not included in current environment (chrome navigation files)
-    const matchedLinks = availableLinks.reduce<(NavItem & { routeMatch: string })[]>((acc, item) => {
-      const match = linksToMatch.find((link) => matchPath(link, item.href!));
-      if (match) {
-        return [...acc, { ...item, routeMatch: match }];
-      }
-
-      return acc;
-    }, []);
-
-    // pre-filter sections data by filter value
-    // re-create all services section data with links avaiable in current environments
-    return filterAllServicesSections(allServicesLinks, filterValue).reduce<AllServicesSection[]>((acc, curr) => {
-      const sectionLinks = curr.links.filter((item) => {
-        return isAllServicesGroup(item)
-          ? item.links.filter(({ href, isExternal }) => isExternal || matchedLinks.find((link) => matchPath(link.routeMatch, href))).length > 0
-          : item.isExternal || matchedLinks.find((link) => matchPath(link.routeMatch, item.href));
-      });
-      if (sectionLinks.length > 0) {
-        return [...acc, { ...curr, links: sectionLinks }];
-      }
-      return acc;
-    }, []);
-    // run hook after data are loaded or filter value changed
-  }, [ready, filterValue]);
+  const linkSections = useMemo(() => filterAllServicesSections(availableSections, filterValue), [ready, filterValue]);
 
   // Provide a flat list of all avaiable links
   const servicesLinks = useMemo(
     () =>
       linkSections
-        .flatMap(({ links }) => links)
+        .flatMap(({ links }) => links as (AllServicesGroup | AllServicesLink)[])
```

Comment:
The option of the link is a `string` is messing this all up. Maybe, it should have been added to the type and if necessary type cast the scenario that can expect it to be a `string`. That way we could remove a lot of casting.

Reply:
Updated all `AllServicesGroup` not to include `string` as option and same goes for the sections.